saleor/graphql/shop/mutations.py

```diff
@@ -169,6 +186,17 @@ def clean_input(cls, _info, _instance, data):
                 )
             if not new_value:
                 data["limit_quantity_per_checkout"] = None
+        if "metadata" in data:
+            metadata = data.pop("metadata", [])
+            metadata_dict = {item["key"]: item["value"] for item in metadata}
+            data["metadata"] = metadata_dict
+
+        if "private_metadata" in data:
+            private_metadata = data.pop("private_metadata", [])
+            private_metadata_dict = {
+                item["key"]: item["value"] for item in private_metadata
+            }
+            data["private_metadata"] = private_metadata_dict
```

Comment:
I think this can be skipped since this is a `BaseMutation` we should be able to use set `support_meta_field=True` in the meta class and use `validate_and_update_metadata` in the `perform_mutation`. See other usages of `validate_and_update_metadata` for reference.